pkg/pod/pod.go

```diff
@@ -300,7 +300,12 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 	if shouldAddReadyAnnotationOnPodCreate(ctx, taskSpec.Sidecars) {
 		podAnnotations[readyAnnotation] = readyAnnotationValue
 	}
-	activeDeadlineSeconds := int64(taskRun.GetTimeout(ctx).Seconds() * deadlineFactor)
+
+	var activeDeadlineSeconds *int64
+	if taskRun.GetTimeout(ctx) != config.NoTimeoutDuration {
+		activeDeadlineSeconds = new(int64)
```

Comment:
Thanks @pritidesai. It looks good to me. Not sure do you need to recreate a new pointer here?

Reply:
thanks @Tomcli, the `var` declaration on line 304 points to a `nil` and dereferencing a `nil` pointer on line 307 will result in runtime panic without using zero valued int64 on line 306. Hope this helps!